assets/src/components/GoogleAnalyticsTracking.js

```diff
@@ -0,0 +1,42 @@
+import React, { Component, } from "react"
+import { withRouter } from 'react-router-dom'
+import GoogleAnalytics from "react-ga"
+
+let ga_initialized = false
+
+const trackPage = (page, options = {}) => {
+  const google_analytics_id = myla_globals.google_analytics_id
+  if (!ga_initialized && google_analytics_id) {
+    ga_initialized = true
+    GoogleAnalytics.initialize(google_analytics_id)
+  }
+  if (google_analytics_id) {
+    GoogleAnalytics.set({ page, ...options, })
+    GoogleAnalytics.pageview(page)
+  }
+}
+
+const GoogleAnalyticsTracking = class extends Component {
```

Comment:
is there any reason why you used the class-based component instead of function based components. The whole react design is based on using functional components using Hooks for handling the side effects

Reply:
I think Andrew found this code from here: https://github.com/react-ga/react-ga/wiki/React-Router-v4-withTracker

The hooks version didn't work for some reason.

Reply:
I'm fine with it at this point. I was surprised that it was not done using functional component since your design is based on using it since beggining

Reply:
Me too. We can change in the future once the higher-priority items are complete.

Reply:
I was previously using a slightly modified version of the link Justin provided but have now re-written it to be hook based